### PR TITLE
Use repo ref versus downloading config file

### PR DESCRIPTION
**Pull request by GeekMasher**

### CODE REVIEW

File: .github/workflows/dependency-review.yml [hunk 1]
```diff
@@ -15,23 +15,25 @@ jobs:
...
+            echo "config=./.github/labeler.yml" >> $GITHUB_STATE
+
           else
             echo "No local configuration file found"
-            echo "Downloading configuration file from advanced-security/reusable-workflows repository"
+            echo "Using configuration file from advanced-security/reusable-workflows repository"
 
-            # download file using gh cli
-            gh api repos/advanced-security/reusable-workflows/contents/.github/dependency-review.yml -q '.content' | base64 -d > .github/dependency-review.yml
+            echo "config=advanced-security/reusable-workflows.github/dependency-review.yml@main" >> $GITHUB_STATE
```
Comment (felickz) [inline .github/workflows/dependency-review.yml:31]:
```suggestion
            echo "config=advanced-security/reusable-workflows/.github/dependency-review.yml@main" >> $GITHUB_STATE
```

typo

File: .github/dependency-review.yml [hunk 2]
```diff
@@ -10,4 +10,23 @@ fail-on-severity: moderate
 fail-on-scopes: runtime
 
 # Here is the list of licenses we want to deny and not use in our software. 
-deny-licenses: GPL-1.0-only,GPL-1.0-or-later,GPL-2.0-only,GPL-2.0-or-later,GPL-3.0-only,GPL-3.0-or-later,AGPL-1.0-only,AGPL-1.0-or-later,AGPL-3.0-only,AGPL-3.0-or-later,LGPL-2.0-only,LGPL-2.0-or-later,LGPL-2.1-only,LGPL-2.1-or-later,LGPL-3.0-only,LGPL-3.0-or-later
+deny-licenses:
```
Comment (felickz) [inline .github/dependency-review.yml:13]: We might consider expanding this to spdx equivalent 

```
RPL-1.1
RPL-1.5
RPSL-1.0
OCLC-2.0
NGPL
ODbL-1.0
MS-RL
OSL-1.0
OSL-1.1
OSL-2.0
OSL-2.1
OSL-3.0
MPL-2.0
MPL-1.0
MPL-1.1
MPL-2.0-no-copyleft-exception
EPL-1.0
EPL-2.0
```

I rather like the allow list approach: https://github.com/ctcampbell/blueoak-to-dependency-review/blob/main/dist/blueoak-licenses.yml

**Review by felickz — APPROVED** [review]
Looks good if you merge in my minor fix

File: .github/dependency-review.yml (hunk 2 above)
Comment (GeekMasher) [reply .github/dependency-review.yml]: Completely valid request and I have set it to use that list now

**Pull request merged**